presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -558,6 +566,51 @@ protected Node visitShowSession(ShowSession node, Void context)
                     identifier("include"));
         }
 
+        private void checkObjectAccessible(Node node, QualifiedObjectName objectName)
+        {
+            if (!isCatalogAccessible(objectName.getCatalogName())) {
+                throw new SemanticException(MISSING_SCHEMA, node, "Catalog '%s' does not exist", objectName.getCatalogName());
+            }
+
+            if (!isSchemaAccessible(objectName.asCatalogSchemaName())) {
+                throw new SemanticException(MISSING_SCHEMA, node, "Schema '%s' does not exist", objectName.getSchemaName());
+            }
+
+            if (!isViewAccessible(objectName) && !isTableAccessible(objectName)) {
```

Comment:
Why isn't this `if (!isRelationAccessible(objectName))` ?
(also, sometimes we say "table" meaning "table or view" (`accessControl.filterTables` ?), so `isTableAccessible` could accept views... but i am not a fan of this approach)

Reply:
The best would be to separate them and have `filterViews`